### PR TITLE
Migrate obs-infraobs-integrations to package-spec v3 #3

### PR DESCRIPTION
**Pull request by shmsr**
Migrate a batch of packages from v1/v2 to v3 package-spec. Used the following scripts and some manual changes to generate the changes in this PR:

yq_hack.zsh
```
#!/bin/bash

set -e
set -u
set -o pipefail
set -x

pushd $1

find data_stream -type f -path "*fields*" -name "*.yml" -exec yq e -i "del .[].release" {} \;
find data_stream -type f -path "*fields*" -name "ecs.yml" -exec yq e -i "unique_by(.name)" {} \;
find data_stream -type f -path "*fields*" -name "*.yml" -exec yq e -i "del .[].fields.[].required" {} \;

yq e -i 'del .release' manifest.yml
yq e -i "with(select(.license != null); .conditions.elastic.subscription = .license) | del .license" manifest.yml

popd

git add -u $1/
git commit -m "[$1]: migration with yq"
```

ecs-update.zsh
```
#!/bin/bash

set -u
set -o pipefail
set -x

echo "-- executing ecs-update for $1 --"

ecs-update -format-version=3.0.0 -fix-dotted-yaml-keys -add-owner-type -owner elastic/obs-infraobs-integrations -v -skip-format=true $1

if (( $? )) 
then
    echo "-- write validation for $1 --"
    pushd $1

    go run ../scripting/write_validation.go

    popd

    # Try again?
    git checkout .
    ecs-update -format-version=3.0.0 -fix-dotted-yaml-keys -add-owner-type -owner elastic/obs-infraobs-integrations -v -skip-format=true $1
fi
```

And another Go program to write the validation.yml by parsing the elastic-package lint errors.

Then I hooked this scripts to run with:

```zsh
$ pkgs=($(cat .github/CODEOWNERS | grep '@elastic/obs-infraobs-integrations' | cut -d ' ' -f 1 | cut -d '/' -f 3 | sort | uniq))

$ for ((idx=16; idx<=30; idx++))
do
scripting/yq_hack.zsh $pkgs[idx]
done
```

```zsh
$ for ((idx=16; idx<=30; idx++))
do
scripting/ecs_update.zsh $pkgs[idx] || break
done
```

Another variation of the above script:

```zsh
$ for ((idx=31; idx<=48; idx++))
do
case $pkgs[idx] in
  "tomcat" | "salesforce" | "redis") echo "-- skip $pkgs[idx] --";;
  "websphere_application_server" | "system" | "statsd_input" | "sql_input" | "springboot" | "rabbitmq" | "postgresql" | "prometheus_input" | "oracle_weblogic" | "php_fpm" | "oracle") echo "-- done $pkgs[idx] --";;
  *) scripting/ecs_update.zsh $pkgs[idx] || break;;
esac
done
```

As I am creating a batch of 18 packages here, I am using a for loop. In case the migration is not possible for some packages, I put it in the first case to skip them.

## Proposed commit message

Migrate packages owned by obs-infraobs-integrations but few couldn't be migrated. Marking them as follows:

- [x] oracle
- [x] oracle_weblogic
- [x] php_fpm
- [x] postgresql
- [x] prometheus_input
- [x] rabbitmq
- [ ] redis
- [x] redisenterprise
- [ ] salesforce
- [x] spring_boot
- [ ] sql_input
- [ ] statsd_input
- [x] system
- [ ] tomcat
- [x] traefik
- [x] vsphere
- [x] websphere_application_server
- [x] zookeeper

Please see the comment  https://github.com/elastic/integrations/pull/8203#issuecomment-1763791793 in this PR itself to understand why some packages couldn't be migrated and also learn about the challenges faced when migrating.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

Relates to #8170
Relates to #8171

### CODE REVIEW

**Comment by shmsr** [comment]
- oracle
    - Found duplicate in fields.yml
        - Improving my script. For some reason right syntax with unique_by is not working with yq. Not sure why?
    - Please review oracle carefully. Changes made in ingest_pipeline as user.roles needed to be normalized to an array
- oracle_weblogic
    - Please review carefully. Changes made in ingest_pipeline as oracle_weblogic.access.host_address / host.ip needed to be normalized to an array
- redis
    - Error: building package failed: resolving external fields failed: can't resolve fields: field os.full cannot be reused at top level
- salesforce
```
    - 2023/10/16 01:33:46 salesforce: Failed: failed modifying ingest pipeline at data_stream/logout_rest/elasticsearch/ingest_pipeline/default.yml: [38:5] unexpected key name
    -       35 |     - ISO8601
    -       36 |     ignore_failure: true
    -       37 | - rename:
    -     > 38 |     field: json.SESSION_TYPE
    -       39 |     target_field: salesforce.logout.session.type
    -                ^
    -       40 |     ignore_failure: true
    -       41 |     ignore_missing: true
    -       42 | - rename:

```
- springboot
    - Review carefully as ingest_pipeline is modified
- sql_input
    - Have set object_type as keyword temporarily. Need to discuss and change that.
- ecs_update used [sql] instead [sql_input] for sql_input
- statsd_input
    - Have set object_type as keyword temporarily. Need to discuss and change that.
- system
    - Carefully review as it a big one
    - Change object to flattened in data_stream/process/fields/fields.yml
- tomcat
    - Error: building package failed: resolving external fields failed: can't resolve fields: field geo.city_name cannot be reused at top level

File: packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -138,9 +138,12 @@ processors:
       field: tmp_timestamp
       patterns:
         - "%{ISO8601_TIMEZONE:event.timezone}$"
-  - rename:
+  - set:
+      field: user.roles
+      value: ["{{{oracle.database_audit.privilege}}}"]
```
Comment (ishleenk17) [inline packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml:143]: Using set instead of rename?
Any pr requisite of package spec 3 ?

File: packages/oracle_weblogic/data_stream/access/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -54,6 +54,11 @@ processors:
     target_field: host.ip
     ignore_missing: true
     ignore_failure: true
+- set:
+    field: host.ip
```
Comment (ishleenk17) [inline packages/oracle_weblogic/data_stream/access/elasticsearch/ingest_pipeline/default.yml:58]: More details on this please? 
Why is this added?

File: packages/postgresql/data_stream/log/_dev/test/pipeline/test-postgresql-11-start-stop-csv.log-expected.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
                 "category": [
                     "database"
                 ],
-                "ingested": "2023-01-19T09:44:52.625481257Z",
```
Comment (ishleenk17) [inline packages/postgresql/data_stream/log/_dev/test/pipeline/test-postgresql-11-start-stop-csv.log-expected.json:12]: There are quite a lot of files in this PR where the expected json is there in the diff due to the timestamp change.
Can we revert those? It adds to the noise in the PR

File: packages/rabbitmq/data_stream/queue/fields/agent.yml [hunk 4]
```diff
@@ -202,7 +202,7 @@
       example: "stretch"
       description: >
         OS codename, if any.
-        
+
```
Comment (ishleenk17) [inline packages/rabbitmq/data_stream/queue/fields/agent.yml:205]: Also, a number of files having such diffs? Are these part of some linting done ?

File: packages/prometheus_input/fields/input.yml [hunk 5]
```diff
@@ -43,3 +43,4 @@
   object_type_mapping_type: "*"
   description: >
```
Comment (ishleenk17) [inline packages/prometheus_input/fields/input.yml:44]: This package has quite a lot of mappings having object_type_mappting_type. 
Dpo they stay intact ?

File: packages/system/data_stream/process/fields/fields.yml [hunk 6]
```diff
@@ -11,7 +11,7 @@
         The full command-line used to start the process, including the arguments separated by space.
       ignore_above: 2048
     - name: env
-      type: object
+      type: flattened
```
Comment (ishleenk17) [inline packages/system/data_stream/process/fields/fields.yml:14]: Please elaborate on why this choice was made ?

File: packages/rabbitmq/data_stream/queue/fields/agent.yml (hunk 4 above)
Comment (shmsr) [reply packages/rabbitmq/data_stream/queue/fields/agent.yml:205]: ecs-update tool also runs elastic-package format: https://github.com/andrewkroh/go-examples/blob/271532fbc14d110ec24cac6b7ef6d0c8ea51812f/ecs-update/ecs-update.go#L340

File: packages/postgresql/data_stream/log/_dev/test/pipeline/test-postgresql-11-start-stop-csv.log-expected.json (hunk 3 above)
Comment (shmsr) [reply packages/postgresql/data_stream/log/_dev/test/pipeline/test-postgresql-11-start-stop-csv.log-expected.json:12]: It'll be a huge manual effort though to filter out if actually changes are there or just timestamp changes. These changes happen as a result of running ecs-update tool.

File: packages/oracle_weblogic/data_stream/access/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (shmsr) [reply packages/oracle_weblogic/data_stream/access/elasticsearch/ingest_pipeline/default.yml:58]: Because an array is required.

Faced this: https://github.com/elastic/elastic-package/blob/main/docs/howto/update_major_package_spec.md#field--is-not-normalized-as-expected-expected-array-found-

Comment (shmsr) [reply packages/oracle_weblogic/data_stream/access/elasticsearch/ingest_pipeline/default.yml:58]: I know this is a hacky way of converting it to an array but I couldn't find a better solution. Is there a better way?

File: packages/system/data_stream/process/fields/fields.yml (hunk 6 above)
Comment (shmsr) [reply packages/system/data_stream/process/fields/fields.yml:14]: If you remember our discussion with Tom earlier:

> * Objects without object_type that seem to fit as "tags" or "labels" use cases have been converted to flattened.
> * Objects without object_type that seem to fit as metrics have been converted to type: double.

So env is similar to tags i.e., key-value pair. So, I have converted it to flattened type.

File: packages/prometheus_input/fields/input.yml (hunk 5 above)
Comment (shmsr) [reply packages/prometheus_input/fields/input.yml:44]: Can't comment on this. Need help here.

File: packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (shmsr) [reply packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml:143]: I replaced `rename` with `set` followed by a `remove`.

The expectation is to have array type for user.roles (similar to: https://github.com/elastic/integrations/pull/8203#discussion_r1360452812)

Is there a better way to achieve this? From the documentation I could come with this only.

File: packages/postgresql/data_stream/log/_dev/test/pipeline/test-postgresql-11-start-stop-csv.log-expected.json (hunk 3 above)
Comment (shmsr) [reply packages/postgresql/data_stream/log/_dev/test/pipeline/test-postgresql-11-start-stop-csv.log-expected.json:12]: https://github.com/elastic/integrations/pull/8170#discussion_r1360485763

Comment (shmsr) [reply packages/postgresql/data_stream/log/_dev/test/pipeline/test-postgresql-11-start-stop-csv.log-expected.json:12]: Here's something that might help in making the review easier: https://github.com/elastic/integrations/issues/8028#issuecomment-1765164454?

File: packages/oracle/changelog.yml [hunk 7]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: 1.22.0
+  changes:
+    - description: Update the package format_version to 3.0.0.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/8194
```
Comment (muthu-mps) [inline packages/oracle/changelog.yml:6]: small correction here, PR link id needs to be updated to `8203`

Comment (shmsr) [reply packages/oracle/changelog.yml:6]: Thank you.

Comment (shmsr) [reply packages/oracle/changelog.yml:6]: Have pushed a fix!

**Comment by shmsr** [comment]
CI is now passing for all packages except spring_boot and I see these kind of errors:

```
18T10:49:05.145586832Z","ExitCode":7,"Output":"  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\ncurl: (7) Failed to connect to springboot port 8090: Connection refused\n"},{"Start":"2023-10-18T10:49:06.252375243Z","ExitCode":7,"Output":"  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\ncurl: 

(7) Failed to connect to springboot port 8090: Connection refused\n"}]}}}

[2023-10-18T10:49:08.144Z] 2023/10/18 10:49:08 DEBUG Wait for healthy containers: c7229a58ab18196
```

So looks like there was a network issue or some service was used that very port (i.e., conflict with another PR). Should be fine on a re-run.

**Comment by shmsr** [comment]
As per this comment: https://github.com/elastic/integrations/issues/8028#issuecomment-1769962013 removed `release: beta` from `oracle` and `redisenterprise`

**Comment by shmsr** [comment]
As https://github.com/elastic/integrations/pull/8206 is upgrading to package-spec v3 for `system`, I have removed the changes related to `system` package in this PR.

https://github.com/elastic/integrations/pull/8206 also adds tags.yml which is not present in this PR. So reverted the changes for `system` package.

File: packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (tommyers-elastic) [reply packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml:143]: 👍 

we should also have `ignore_failure: true` here though please.

Comment (tommyers-elastic) [reply packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml:143]: i see it wasn't there in the original rename processor - but i think it's good there just as a failsafe.

File: packages/oracle/data_stream/database_audit/fields/fields.yml [hunk 8]
```diff
@@ -89,11 +89,6 @@
       description: >
         The related OS user.
 
-    - name: os_userid
```
Comment (tommyers-elastic) [inline packages/oracle/data_stream/database_audit/fields/fields.yml:92]: what's the reason for this removal?

Comment (tommyers-elastic) [reply packages/oracle/data_stream/database_audit/fields/fields.yml:92]: oh it's a dup of the one directly above - nevermind.

File: packages/oracle/data_stream/sysmetric/manifest.yml [hunk 9]
```diff
@@ -1,6 +1,5 @@
 title: "Sysmetric related metrics."
 type: metrics
-release: beta
```
Comment (tommyers-elastic) [inline packages/oracle/data_stream/sysmetric/manifest.yml:3]: is this an intentional change? perhaps we should make the change from this DS to go GA in another PR?

File: packages/oracle/data_stream/performance/manifest.yml [hunk 10]
```diff
@@ -1,6 +1,5 @@
 title: "Oracle performance metrics"
 type: metrics
-release: beta
```
Comment (tommyers-elastic) [inline packages/oracle/data_stream/performance/manifest.yml:3]: is this an intentional change? perhaps we should make the change from this DS to go GA in another PR?

File: packages/oracle/data_stream/memory/manifest.yml [hunk 11]
```diff
@@ -1,6 +1,5 @@
 title: "Memory metrics"
 type: metrics
-release: beta
```
Comment (tommyers-elastic) [inline packages/oracle/data_stream/memory/manifest.yml:3]: is this an intentional change? perhaps we should make the change from this DS to go GA in another PR?

File: packages/oracle/data_stream/system_statistics/manifest.yml [hunk 12]
```diff
@@ -1,6 +1,5 @@
 title: "System Statistics"
 type: metrics
-release: beta
```
Comment (tommyers-elastic) [inline packages/oracle/data_stream/system_statistics/manifest.yml:3]: same here?

File: packages/oracle/data_stream/tablespace/manifest.yml [hunk 13]
```diff
@@ -1,6 +1,5 @@
 title: "Oracle tablespace metrics"
 type: metrics
-release: beta
```
Comment (tommyers-elastic) [inline packages/oracle/data_stream/tablespace/manifest.yml:3]: and here

File: packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml [hunk 14]
```diff
@@ -138,9 +138,12 @@ processors:
...
+      ignore_empty_value: true
```
Comment (shmsr) [inline packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml:144]:
```suggestion
      ignore_empty_value: true
      ignore_failure: true
```

File: packages/oracle_weblogic/data_stream/access/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (tommyers-elastic) [reply packages/oracle_weblogic/data_stream/access/elasticsearch/ingest_pipeline/default.yml:58]: there is `append` processor - but it's no better than this.

File: packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (shmsr) [reply packages/oracle/data_stream/database_audit/elasticsearch/ingest_pipeline/default.yml:143]: yeah, sure.

File: packages/oracle/data_stream/performance/manifest.yml (hunk 10 above)
Comment (shmsr) [reply packages/oracle/data_stream/performance/manifest.yml:3]: https://github.com/elastic/integrations/pull/8203#issuecomment-1770274231

Sure, I'll revert the related commit.

File: packages/postgresql/docs/README.md [hunk 15]
```diff
@@ -65,7 +65,7 @@ persistent connections, so enable with care.
 | error.code | Error code describing the error. | keyword |
 | error.id | Unique identifier for the error. | keyword |
 | error.message | Error message. | match_only_text |
-| event.category | Event category (e.g. database) | keyword |
```
Comment (tommyers-elastic) [inline packages/postgresql/docs/README.md:68]: which description is more relevant here?

File: packages/postgresql/validation.yml [hunk 16]
```diff
@@ -0,0 +1,5 @@
+errors:
+  exclude_checks:
+    - SVR00001
```
Comment (tommyers-elastic) [inline packages/postgresql/validation.yml:3]: what is 00001?

File: packages/prometheus_input/fields/input.yml (hunk 5 above)
Comment (tommyers-elastic) [reply packages/prometheus_input/fields/input.yml:44]: how do you mean @ishleenk17 ? i can't see any changes to the mappings in this package?

File: packages/redisenterprise/data_stream/node/manifest.yml [hunk 17]
```diff
@@ -1,6 +1,5 @@
 title: "node"
 type: metrics
-release: beta
```
Comment (tommyers-elastic) [inline packages/redisenterprise/data_stream/node/manifest.yml:3]: same comment here about taking the DS GA.

File: packages/sql_input/fields/input.yml [hunk 18]
```diff
@@ -23,3 +23,4 @@
     - name: metrics.*
       description: Default mapping for metric fields. You need to add your own custom mappings when storing other kinds of information.
       type: object
+      object_type: keyword
```
[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply packages/sql_input/fields/input.yml:26]: this mapping isn't doing anything at all - i think the best bet would be to remove entirely? 

the other sub-metric types have mappings already.

File: packages/postgresql/validation.yml (hunk 16 above)
Comment (shmsr) [reply packages/postgresql/validation.yml:3]: CodeKibanaDashboardWithQueryButNoFilter

File: packages/statsd_input/fields/input.yml [hunk 19]
```diff
@@ -8,3 +8,4 @@
 - name: statsd.*.*
   description: Default mapping for metric fields. You need to add your own custom mappings when storing other kinds of information.
   type: object
+  object_type: keyword
```
[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply packages/statsd_input/fields/input.yml:11]: same here - there currently is no mapping. what is the intention behind including it?

File: packages/oracle/data_stream/performance/manifest.yml (hunk 10 above)
Comment (shmsr) [reply packages/oracle/data_stream/performance/manifest.yml]: done.

File: packages/postgresql/docs/README.md (hunk 15 above)
Comment (shmsr) [reply packages/postgresql/docs/README.md:68]: IMO, ECS definition is better because there's an allowed list of values: https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html

Comment (tommyers-elastic) [reply packages/postgresql/docs/README.md:68]: do we conform to the allowed values here?

File: packages/statsd_input/fields/input.yml (hunk 19 above)
[the remark this answers is not in the document]
Comment (shmsr) [reply packages/statsd_input/fields/input.yml:11]: same goes here too i'm not wrong

https://github.com/elastic/integrations/pull/8203#discussion_r1365474190

File: packages/sql_input/fields/input.yml (hunk 18 above)
[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply packages/sql_input/fields/input.yml:26]: ok no problem - what type of data do you typically see in the base metrics fields? numerical? `double` ?

**Comment by shmsr** [comment]
spring_boot ci is failing for some reason. not able to get what's happening. some curl is failing when trying to hit a port where spring_boot is supposed to run; returning "connection refused".

File: packages/spring_boot/_dev/deploy/docker/Dockerfile [hunk 20]
```diff
@@ -1,9 +1,8 @@
...
-RUN mkdir -p /build
-RUN chmod -R 777 /build
+RUN mkdir -p /build && chmod -R 777 /build
 WORKDIR /build
 COPY ./spring_boot/ /build/spring_boot/
 WORKDIR /build/spring_boot
-RUN mvn package -q
-HEALTHCHECK --interval=1s --retries=90 CMD curl -f http://actuator:actuator@springboot:8090/actuator/info
+RUN mvn package -q -DskipTests
+HEALTHCHECK --interval=5s --retries=5 CMD curl -f http://springboot:8090/actuator/health || exit 1
```
Comment (shmsr) [inline packages/spring_boot/_dev/deploy/docker/Dockerfile:7]: Wrote a better Dockerfile:

- Reduce layers (by combining `RUN`)
- -DskipTests skips tests — saves time and resources by not doing an unnecessary op
- Retry should preferably have gaps for application that takes time to come up. Frequent polling is wasteful. Also, doing the ideal way of health checking a spring boot service (see: [1], [2])

[1]: https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.endpoints
[2]: https://stackoverflow.com/questions/57515333/how-to-do-a-health-check-of-a-spring-boot-application-running-in-a-docker-contai


Now, with a better health check, connection refused errors on the port where Spring Boot was running have been gone.

Comment (shmsr) [reply packages/spring_boot/_dev/deploy/docker/Dockerfile:7]: Related: https://github.com/elastic/integrations/pull/8203#issuecomment-1771215972

File: packages/spring_boot/manifest.yml [hunk 21]
```diff
@@ -1,15 +1,17 @@
...
+version: "0.13.0"
 description: This Elastic integration collects logs and metrics from Spring Boot integration.
 type: integration
 categories:
   - observability
   - java_observability
 conditions:
-  kibana.version: ^8.2.0
+  kibana:
+    version: ^8.9.0
```
Comment (shmsr) [inline packages/spring_boot/manifest.yml:12]: Bumped the stack version to verify something in CI. 

CI has been failing for spring_boot from the very beginning on every single CI fresh trigger. While debugging found couple of issues:

1. Related to bad health check for which I have pushed changes in Dockerfile
2. System tests are still failing with the log:

```
3/10/19 22:46:40 DEBUG found 0 hits in logs-spring_boot.audit_events-ep data stream: index_not_found_exception: no such index [logs-spring_boot.audit_events-ep] Status=404
```

When debugged locally, found that:

- 8.9.x and above stack: system tests pass
- 8.8.x stack: system tests fail
- 8.2.0 stack: system tests fail

8.2.0 is the stack used for testing in CI to test this package based on the minimum kibana.version mentioned in manifest.yml.

From ES logs, found this:

```
{"@timestamp":"2023-10-19T17:18:49.452Z", "log.level": "WARN", "message":"caught exception while handling client http traffic, closing connection Netty4HttpChannel{localAddress=/192.168.0.3:9200, remoteAddress=/192.168.0.1:63712}", "ecs.version": "1.2.0","service.name":"ES_ECS","event.dataset":"elasticsearch.server","process.thread.name":"elasticsearch[e1afeb61b728][transport_worker][T#6]","log.logger":"org.elasticsearch.http.AbstractHttpServerTransport","elasticsearch.cluster.uuid":"J5KbFaAFTuiJ1-4qINf14Q","elasticsearch.node.id":"KsmKSXd4TGmdB4jAKL0gUg","elasticsearch.node.name":"e1afeb61b728","elasticsearch.cluster.name":"elasticsearch","error.type":"io.netty.handler.codec.DecoderException","error.message":"javax.net.ssl.SSLHandshakeException: Received fatal alert: bad_certificate","error.stack_trace":"io.netty.handler.codec.DecoderException: javax.net.ssl.SSLHandshakeException: Received fatal alert: bad_certificate\n\tat io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:480)\n\tat io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:279)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)\n\tat io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)\n\tat io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)\n\tat io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)\n\tat io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:623)\n\tat io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:586)\n\tat io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\nCaused by: javax.net.ssl.SSLHandshakeException: Received fatal alert: bad_certificate\n\tat java.base/sun.security.ssl.Alert.createSSLException(Alert.java:131)\n\tat java.base/sun.security.ssl.Alert.createSSLException(Alert.java:117)\n\tat java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:358)\n\tat java.base/sun.security.ssl.Alert$AlertConsumer.consume(Alert.java:293)\n\tat java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:204)\n\tat java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)\n\tat java.base/sun.security.ssl.SSLEngineImpl.decode(SSLEngineImpl.java:736)\n\tat java.base/sun.security.ssl.SSLEngineImpl.readRecord(SSLEngineImpl.java:691)\n\tat java.base/sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:506)\n\tat java.base/sun.security.ssl.SSLEngineImpl.unwrap(SSLEngineImpl.java:482)\n\tat java.base/javax.net.ssl.SSLEngine.unwrap(SSLEngine.java:679)\n\tat io.netty.handler.ssl.SslHandler$SslEngineType$3.unwrap(SslHandler.java:295)\n\tat io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1342)\n\tat io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1235)\n\tat io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1284)\n\tat io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:510)\n\tat io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:449)\n\t... 16 more\n"}
```

So pushed 8.9.0 to check if system tests are passing in CI too. It does.

Comment (tommyers-elastic) [reply packages/spring_boot/manifest.yml:12]: thanks for figuring this out. i'm happy to bump the kibana.version.

File: packages/sql_input/fields/input.yml (hunk 18 above)
[the remark this answers is not in the document]
Comment (aliabbas-elastic) [reply packages/sql_input/fields/input.yml:26]: Please refer the [sample_event](https://github.com/elastic/integrations/blob/main/packages/sql_input/sample_event.json#L62) just for your reference. Most of them are numerical fields but this would again depend on the runtime fields as the response may change depending upon the query applied.

[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply packages/sql_input/fields/input.yml:26]: in that case i'm not sure it makes sense for us to add any mapping. it's impossible for us to choose a sensible type here.

[the remark this answers is not in the document]
Comment (shmsr) [reply packages/sql_input/fields/input.yml:26]: Right. Let me know what to do here.

[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply packages/sql_input/fields/input.yml:26]: decided to pull these changes out into a seperate PR to unblock this one

File: packages/statsd_input/fields/input.yml (hunk 19 above)
[the remark this answers is not in the document]
Comment (tommyers-elastic) [reply packages/statsd_input/fields/input.yml:11]: these changes are going into a seperate PR to unblock this one.

**Review by tommyers-elastic — APPROVED** [review]
